src/js/models/layerstructure.js

```diff
@@ -895,9 +895,7 @@ define(function (require, exports, module) {
             }, this);
         }.bind(this));
 
-        return this.mergeDeep({
-            layers: nextLayers
-        });
+        return this.set("layers", this.layers.merge(nextLayers));
```

Comment:
`mergeDeep` will try to merge two iterables, instead of replacing. In our case, Layer#dropShadows and Layer#innerShadows are iterables, and they will be wrongly merged into a super set of the old and new shadows.